homeassistant/components/huisbaasje/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow for Huisbaasje integration."""
+import logging
+
+from huisbaasje import Huisbaasje, HuisbaasjeConnectionException, HuisbaasjeException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class HuisbaasjeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Huisbaasje."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return await self._show_setup_form(user_input)
+
+        errors = {}
+
+        try:
+            user_id = await self._validate_input(user_input)
+
+            _LOGGER.info("Input for Huisbaasje is valid!")
+
+            # Set user id as unique id
+            existing_config_entry = await self.async_set_unique_id(user_id)
+
+            # Abort if entry already exists
+            if existing_config_entry is not None:
+                return self.async_abort(reason="already_configured")
+            if await self._entry_exists(user_id):
+                return self.async_abort(reason="already_configured")
+
+            # Create entry
+            return self.async_create_entry(
+                title=user_input[CONF_USERNAME],
+                data={
+                    CONF_ID: user_id,
+                    CONF_USERNAME: user_input[CONF_USERNAME],
+                    CONF_PASSWORD: user_input[CONF_PASSWORD],
+                },
+            )
+        except HuisbaasjeConnectionException as exception:
+            _LOGGER.warning(exception)
+            errors["base"] = "connection_exception"
+        except HuisbaasjeException as exception:
```

Comment:
Is this exception only raised for bad auth, or also for other errors?

Reply:
It also for other errors, for example if the result is not json or any other error code